apm-sniffer/apm-sdk-plugin/mongodb-2.x-plugin/src/main/java/org/skywalking/apm/plugin/mongodb/v2/MongoDBCollectionMethodInterceptor.java

```diff
@@ -0,0 +1,90 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Project repository: https://github.com/OpenSkywalking/skywalking
+ */
+
+package org.skywalking.apm.plugin.mongodb.v2;
+
+import com.mongodb.AggregationOutput;
+import com.mongodb.CommandResult;
+import com.mongodb.WriteResult;
+import java.lang.reflect.Method;
+import org.skywalking.apm.agent.core.context.ContextCarrier;
+import org.skywalking.apm.agent.core.context.ContextManager;
+import org.skywalking.apm.agent.core.context.tag.Tags;
+import org.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.skywalking.apm.agent.core.context.trace.SpanLayer;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceConstructorInterceptor;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.skywalking.apm.network.trace.component.ComponentsDefine;
+
+/**
+ * {@link MongoDBCollectionMethodInterceptor} intercept method of {@link com.mongodb.DBCollection#find()}
+ * or{@link com.mongodb.DBCollectionImpl#insert}.... record the mongoDB host, operation name ...
+ */
+
+public class MongoDBCollectionMethodInterceptor implements InstanceMethodsAroundInterceptor, InstanceConstructorInterceptor {
+
+    private static final String DB_TYPE = "MongoDB";
+
+    private static final String MONGO_DB_OP_PREFIX = "MongoDB/";
+
+    @Override public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments,
+        Class<?>[] argumentsTypes, MethodInterceptResult result) throws Throwable {
+
+        String remotePeer = (String)objInst.getSkyWalkingDynamicField();
+        String opertaion = null;
+        opertaion = method.getName();
+        AbstractSpan span = ContextManager.createExitSpan(MONGO_DB_OP_PREFIX + opertaion, new ContextCarrier(), remotePeer);
+        span.setComponent(ComponentsDefine.MONGODB);
+        Tags.DB_TYPE.set(span, DB_TYPE);
+        SpanLayer.asDB(span);
+
+    }
+
+    @Override public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments,
+        Class<?>[] argumentsTypes, Object ret) throws Throwable {
+        AbstractSpan activeSpan = ContextManager.activeSpan();
+        CommandResult cresult = null;
+        if (ret instanceof WriteResult) {
+            WriteResult wresult = (WriteResult)ret;
+            cresult = wresult.getCachedLastError();
+        } else if (ret instanceof AggregationOutput) {
+            AggregationOutput aresult = (AggregationOutput)ret;
+            cresult = aresult.getCommandResult();
+        }
+        if (null != cresult && !cresult.ok()) {
+            activeSpan.tag("CommandError", cresult.getErrorMessage());

```

Comment:
I think you should log the exception

Reply:
I am very worried about this, is this tag necessary? What is the diffetent between this error message and log exception? I need a certain answer.

Reply:
@wu-sheng  this tag is not necessary. the error message will display in `ErrorException`, 
@lytscu   I this tag should  removed.